MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/MathRenderer.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import base64
+import io
+import re
+
+from matplotlib import pyplot as plt
+
+
+class MathRenderer:
+    # Cache mapping the raw LaTex expression to its rendered image form
+    cache = {}
+
+    # Ignore the following expression
+    ignores = {r"\mathbf{q}": "q"}
+
+    def __init__(self, text: str) -> None:
+        self.raw_text = text
+
+    @staticmethod
+    def replace_ignored(text: str) -> str:
+        return MathRenderer.ignores[text]
+
+    @staticmethod
+    def ignore(text: str) -> bool:
+        return text in MathRenderer.ignores
+
+    @staticmethod
+    def containsMultiLineBlockExpression(text: str) -> bool:
+        return text in {".. math::", r"<br />.. math::<br />"}
+
+    @staticmethod
+    def containsBlockExpression(text: str) -> bool:
+        return text.startswith(".. math:")
+
+    @staticmethod
+    def containsInlineExpressions(text: str) -> bool:
+        pattern = r"(:math:`.*?`)"
+        substrings = re.split(pattern, text)
+        return len(substrings) > 1
+
+    @staticmethod
+    def processBlockExpression(text: str) -> list[tuple[str, bool]]:
+        return [(text[len(".. math:: ") :], True)]
```

Comment:
As above, we can use pattern matching:
```suggestion
    def processBlockExpression(text: str) -> list[tuple[str, bool]]:
        pattern = ".. math::\s*(.+)"
        
        for match in re.finditer(pattern, text):
            text.replace(match[0], self.render(match[1]))
```